File: Chapter 1/Problem 10/Ch1_Problem10.java

import java.util.ArrayList;
import java.util.List;

public class Ch1_Problem10 {

    static List<Long> Factorials = new ArrayList<>() {
        {
            add((long) 1);
            add((long) 1);
        }
    };

    public static long Factorial(int n) {
        for (int i = Factorials.size(); i <= n; i++) {
            Factorials.add(Factorials.get(i - 1) * i);
        }

        return Factorials.get(n);
    }

    public static double sumInverseFactorials(int n) {
        if (n == 1) {
            return 1;
        }
        return (1 / Factorial(n)) + sumInverseFactorials(n - 1);
    }

}

File: Chapter 1/Problem 10/Ch1_Problem10.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<unsigned long long> Factorials = {1, 1};

unsigned long long Factorial(int n)
{
    for (int i = Factorials.size(); i <= n; i++)
        Factorials.push_back(Factorials[(i - 1)] * i);

    return Factorials[n];
}

long double sumInverseFactorials(int n)
{
    if (n == 1)
        return 1;
    return (1 / Factorial(n)) + sumInverseFactorials(n - 1);
}